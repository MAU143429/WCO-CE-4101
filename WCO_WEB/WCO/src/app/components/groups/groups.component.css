body{
  justify-content: center;
  align-items: center;
  background-color: white;
  height: auto;
  min-height: 100%;

}
.container{
  height: auto;
  max-width: 1700px;
  width: 100%;
  background-color: rgba(255, 255, 255, 0.295);
  padding: 25px 30px;
  border-radius: 5px;
  box-shadow: 0 5px 10px rgba(0,0,0,0.20);
}


.input-box{
  display: flex;
  margin-bottom: 15px;
  margin-left: 7%;
  width: 90%;
  font-size: 20px;
}

.title{
  width: 40%;
  font-size: 30px;
  font-weight: 500;
  margin-top: 1%;

}
.input-box .code{
  height: 45px;
  width: 15%;
  margin-top: 1%;
  margin-left: 2%;
  font-size: 20px;
  border-radius: 5px;
  border: 1px solid #ccc;
  border-bottom-width: 2px;
  transition: all 0.3s ease;
  color: black;
}
.input-box .inbtn{
  height: 45px;
  width: 11%;
  margin-top: 1%;
  margin-left: 2%;
  font-size: 20px;
  border-radius: 5px;
  border: 1px solid #ccc;
  border-bottom-width: 2px;
  transition: all 0.3s ease;
  color: white;
  background: linear-gradient(135deg, #1277D9, #7bb0e6, #1277D9);
}

.viewbtn{
  margin-left: 20%;
  font-size: 17px;
  border-radius: 5px;
  border: 1px solid #ccc;
  border-bottom-width: 2px;
  transition: all 0.3s ease;
  color: white;
  background: linear-gradient(135deg, #1277D9, #7bb0e6, #1277D9);
}

#groups {
  font-family: Arial, Helvetica, sans-serif;
  border-collapse: collapse;
  width: 100%;
}

#groups td, #groups th {
  border: 1px solid #ddd;
  padding: 8px;
}

#groups tr:nth-child(even){background-color: #f2f2f2;}

#groups tr:hover {background-color: #ddd;}

#groups th {
  padding-top: 12px;
  padding-bottom: 12px;
  text-align: left;
  background-color: #1277D9;
  color: white;
  text-align: center;
}

::ng-deep snack-bar-container.red-snackbar {
  background: #F44336;
  color: white;
}
::ng-deep snack-bar-container.green-snackbar {
  background: rgb(15, 206, 15);
  color: white;
}

