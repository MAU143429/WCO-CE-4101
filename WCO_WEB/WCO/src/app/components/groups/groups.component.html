<!DOCTYPE html>
<html lang="en" dir="ltr">
  <head>
    <meta charset="UTF-8">
    <!---Referencia al archivo css que contiene el estilo--->
    <link rel="stylesheet" href="groups.component.css">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
   </head>

<body>
  <app-navbar></app-navbar>
  <br>
  <br>
  <br>
  <br>
  <div class="input-box">
    <div class="title">Mis grupos</div>
    <input style="margin-left: 37%;" class="code" type="text" maxlength="12" [(ngModel)]= "groupCode" [ngModelOptions]="{standalone: true}" placeholder="Ingrese ID de grupo">
    <input class="inbtn" type="submit" value="Unirme Grupo" (click)="joinGroup()">
    <input class="inbtn" type="submit" value="Crear Grupo" routerLink="/create-groups">
  </div>
  <div class="container">
    <div class="content">

      <table id="groups" class="table table-bordered mt-4 mytable">
        <tr class="table-primary mytable">
          <th>Nombre del Grupo</th>
          <th>ID de invitaci√≥n</th>
          <th>Grupo de torneo</th>
          <th>Detalles</th>
        </tr>
      <tr *ngFor="let data of groupsData">
        <td>{{data.name}}</td>
          <td>{{data.gId}}</td>
          <td>{{data.tId}}</td>
          <td><button class="viewbtn" (click)="showGroup(data.gId)" routerLink="/group-details" >Ver Grupo</button></td>
      </tr>
      </table>
    </div>
  </div>
</body>
</html>
