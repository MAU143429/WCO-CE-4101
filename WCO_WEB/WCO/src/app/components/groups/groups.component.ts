import { Component, OnInit } from '@angular/core';
import { MatSnackBar } from '@angular/material/snack-bar';
import { JoinGroup } from 'src/app/model/join-group';
import { AccountService } from 'src/app/services/account.service';
import { Group } from '../../interface/group';

@Component({
  selector: 'app-groups',
  templateUrl: './groups.component.html',
  styleUrls: ['./groups.component.css'],
})
export class GroupsComponent implements OnInit {
  status = true;
  groupCode = '';
  groupList: any;
  newMember: JoinGroup = new JoinGroup();
  groupsData: Group[] = [];
  constructor(
    private service: AccountService,
    private _snackBar: MatSnackBar
  ) {}

  /**
   * Este metodo permite realizar un pequeño delay
   * @param ms el tiempo del delay en ms
   */
  async delay(ms: number) {
    await new Promise<void>((resolve) => setTimeout(() => resolve(), ms)).then(
      () => console.log('fired')
    );
  }

  ngOnInit(): void {
    this.updateGroups();
  }

  joinGroup() {
    this.verifyGroupData();
    // Verificar que se lleno el campo
    this.service
      .getGroupByID(this.groupCode)
      .subscribe((data) => (this.groupList = data));

    this.delay(50).then(() => {
      if (this.groupCode == '') {
        this.openError(
          'Falta al menos uno de los espacios requeridos!',
          'Intente de nuevo'
        );
      } else if (this.groupList.length == 0) {
        // Verificar que el grupo exista
        this.openError(
          'Error en el código ingresado',
          'No hay ningún grupo asociado a este código'
        );
      } else if (!this.status) {
        // Verificar que el usuario no este en otro grupo del mismo torneo
        this.openError(
          'Error al unirse al grupo',
          'Usted ya se encuentra en otro grupo del mismo torneo'
        );
      } else {
        this.newMember.acc_nick = localStorage.getItem('nickname');
        this.newMember.acc_email = localStorage.getItem('email');
        this.newMember.gId = this.groupCode;
        // Agregarlo al grupo
        this.service
          .joinGroup(this.newMember)
          .subscribe((data) => console.log(data));
        this.delay(50).then(() => {
          this.updateGroups();
        });
      }
    });
  }

  verifyGroupData() {
    this.groupsData.forEach((value: any) => {
      if (value.tId == this.groupCode.substring(0, 6)) {
        this.status = false;
      }
    });
  }

  updateGroups() {
    this.service
      .getUserGroups(
        localStorage.getItem('nickname'),
        localStorage.getItem('email')
      )
      .subscribe((data) => (this.groupsData = data));
  }

  /**
   * Metodo para mostrar alerta de error por 2 segundos
   * @param message1 Mensaje de error
   * @param message2 Mensaje para cerrar alerta
   */
  openError(message: string, message2: string) {
    this._snackBar.open(message, message2, {
      duration: 2000,
      horizontalPosition: 'center',
      verticalPosition: 'top',
      panelClass: 'red-snackbar',
    });
  }

  /**
   * Metodo para mostrar alerta de exito por 2 segundos
   * @param message1 Mensaje de exito
   * @param message2 Mensaje para cerrar alerta
   */
  openSuccess(message1: string, message2: string) {
    this._snackBar.open(message1, message2, {
      duration: 2000,
      horizontalPosition: 'center',
      verticalPosition: 'top',
      panelClass: 'green-snackbar',
    });
  }

  showGroup(gId: any) {
    localStorage.setItem('gId', gId);
  }
}
