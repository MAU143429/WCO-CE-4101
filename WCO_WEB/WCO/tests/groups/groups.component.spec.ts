import { ComponentFixture, TestBed } from '@angular/core/testing';

import { GroupsComponent } from '../../src/app/components/groups/groups.component';

describe('GroupsComponent', () => {
  let component: GroupsComponent;
  let fixture: ComponentFixture<GroupsComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ GroupsComponent ]
    })
    .compileComponents();

    fixture = TestBed.createComponent(GroupsComponent);
    component = fixture.componentInstance;
    compiled = fixture.nativeElement;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  test('debe de hace match con el snapshot', () => {
    expect(compiled.innerHTML).toMatchSnapshot();
  });
});
