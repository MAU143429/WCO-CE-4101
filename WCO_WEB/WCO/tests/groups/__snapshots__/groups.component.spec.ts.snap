// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupsComponent debe de hace match con el snapshot 1`] = `
"<html lang=\\"en\\" dir=\\"ltr\\"><head><meta charset=\\"UTF-8\\"><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\"></head><body><app-navbar></app-navbar><br><br><br><br><div class=\\"input-box\\"><div class=\\"title\\">Mis grupos</div><input type=\\"text\\" placeholder=\\"Ingrese ID de grupo\\" class=\\"code\\" style=\\"margin-left: 37%;\\"><input type=\\"submit\\" value=\\"Unirme Grupo\\" class=\\"inbtn\\"><input type=\\"submit\\" value=\\"Crear Grupo\\" routerlink=\\"/create-groups\\" class=\\"inbtn\\"></div><div class=\\"container\\"><div class=\\"content\\"><table id=\\"groups\\" class=\\"table table-bordered mt-4 mytable\\"><tr class=\\"table-primary mytable\\"><th>Nombre del Grupo</th><th>ID de invitaci√≥n</th><th>Grupo de torneo</th><th>Detalles</th></tr><tr><td>Los Compas</td><td>5423JKD3D3F55FS4</td><td>Champions League</td><td><button routerlink=\\"/group-details\\" class=\\"viewbtn\\">Ver Grupo</button></td></tr><!--bindings={
  \\"ng-reflect-ng-for-of\\": \\"[object Object]\\"
}--></table></div></div></body></html>"
`;
